perf/impl/go-libp2p/v0.31/Dockerfile

```diff
@@ -0,0 +1,14 @@
+# Build Go Binary
+FROM golang:1.20-alpine AS builder
```

Comment:
We should be building with Go 1.21.

Reply:
Just to confirm, it should be version from `go.mod` + 1, right?

Reply:
Assuming we stick with the 2 Go version policy, yes.